app/src/main/java/com/keylesspalace/tusky/components/viewthread/ViewThreadViewModel.kt

```diff
@@ -95,36 +98,49 @@ class ViewThreadViewModel @Inject constructor(
     }
 
     fun loadThread(id: String) {
+        _uiState.value = ThreadUiState.Loading
+
         viewModelScope.launch {
+            Log.d(TAG, "Finding status with: $id")
             val contextCall = async { api.statusContext(id) }
-            val statusCall = async { api.statusAsync(id) }
-
-            val contextResult = contextCall.await()
-            val statusResult = statusCall.await()
+            val timelineStatus = db.timelineDao().getStatus(id)
 
-            val status = statusResult.getOrElse { exception ->
-                _uiState.value = ThreadUiState.Error(exception)
-                return@launch
+            val detailedStatus = if (timelineStatus != null) {
+                Log.d(TAG, "Loaded status from local timeline")
+                val viewDatum = timelineStatus.toViewData(gson) as StatusViewData.Concrete
+                viewDatum.copy(isDetailed = true)
```

Comment:
instead of copying, pass the isDetailed param down to the constructor?

Reply:
You mean like `timelineStatus.toViewData(gson, isDetailed = true)`, and modify `toViewData` to take the extra parameter?

Or something else?

Reply:
yes exactly, saves one object allocation

Reply:
Done.